<?xml version="1.0"?>
<launch>

    <!-- roslaunch amcl_hybrid amcl_launch.launch 2>&1 | grep -v 'TF_REPEATED_DATA\|binarydeb' -->

    <arg name="environment" default="long_gallery"/>
    <!-- 
    Available environments:
        1. mocap
        2. long_hallway
        3. long_gallery
    -->

    <!-- 0. Set simulation time to synchronize bag files and node data-->
	<param name="/use_sim_time" type="bool" value="true"/>

    <!-- 2. Launch Rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find run_amcl)/test_amcl.rviz" />

    <!-- 3. Launch Map -->
    <arg name="map_file" default="$(arg environment).yaml" />
    <arg name="map_path" default="$(find amcl_hybrid)/../../../maps/" />
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_path)$(arg map_file)" />

    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find tiago_description)/robots/tiago.urdf.xacro'" />

    <!-- 4. Isaac Republish Node -->
    <rosparam file="$(find amcl_hybrid)/cfg/odom_params.yaml" command="load" />
    <node pkg="amcl_hybrid" type="isaac_odom_publisher" name="isaac_odom_publisher" output="screen"/>

    <!-- 5. Record Rosbag -->
    <!-- <node name="record" pkg="rosbag" type="record" args="-O /home/neathle/TFM/rosbags/latest.bag /rgb /amcl_hybrid/camera_info /odom /scan /tf /tf_static /clock /map" /> -->
    <node name="record" pkg="rosbag" type="record" args="-O /home/neathle/TFM/rosbags/latest.bag /cmd_vel" />

</launch>